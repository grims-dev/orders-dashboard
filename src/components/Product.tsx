import React from "react";
import styled, { StyledComponent } from "styled-components";
import pending from "../assets/images/pending.png";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faTimesCircle } from "@fortawesome/free-solid-svg-icons";

let midwayBreakpoint = "850px";
console.log(midwayBreakpoint);
type productBreakpoint = {
  breakpoint: string;
};
const StyledProduct: StyledComponent<"div", any, {}, never> = styled.div`
  padding: 45px 10px 25px;
  margin-bottom: 20px;
  border: 1px solid #999;
  border-radius: 10px;
  color: #999;
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  overflow: hidden;
  @media (min-width: ${midwayBreakpoint}) {
    padding: 10px 25px 10px 45px;
    flex-direction: row;
  }
  .status-border {
    height: 20px;
    width: 100%;
    display: block;
    position: absolute;
    left: 0;
    top: 0;
    @media (min-width: ${midwayBreakpoint}) {
      height: 100%;
      width: 20px;
    }
    &.status-readytotry {
      background: var(--status-readytotry);
    }
    &.status-ontheway {
      background: var(--status-ontheway);
    }
    &.status-inthequeue {
      background: var(--status-inthequeue);
    }
    &.status-outofstock {
      background: var(--status-outofstock);
    }
  }
  .shoe-thumbnail {
    border-radius: 10px;
    width: 100px;
    max-width: 100%;
    height: auto;
    max-height: 100%;
  }
  .shoe-name {
    color: #fefefe;
    margin: 10px;
    font-size: 20px;
    line-height: 100%;
    @media (min-width: ${midwayBreakpoint}) {
      margin: 10px 100px 10px 30px;
      max-width: 120px;
    }
  }
  .shoe-info {
    font-size: 20px;
    line-height: 100%;
    margin: 10px;
    flex-grow: 1;
    flex-basis: 0;
    align-self: start;
    @media (min-width: ${midwayBreakpoint}) {
      align-self: inherit;
    }
  }
  .initials {
    color: #fefefe;
    font-size: 18px;
    border: 1px solid #999;
    border-radius: 10px;
    width: 50px;
    height: 50px;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .btn-remove {
    background-color: transparent;
    border: none;
    color: #fefefe;
    padding: 10px;
    &:hover {
      color: red;
    }
  }
`;

export interface ProductUI {
  position: number;
  product: {
    id: number;
    product_name: string;
    category: string;
    size: string;
    colour: string;
    status: string;
    customers_initials: string;
    product_image?: string;
  };
  handleProductRemove: (id: number, productname: string) => void;
}

const Product = (props: ProductUI) => (
  <StyledProduct data-test="product">
    <div
      className={`status-border status-${props.product.status}`}
      aria-hidden="true"
    />

    <img
      src={props.product.product_image ? props.product.product_image : pending}
      alt={props.product.product_name}
      className="shoe-thumbnail"
      width="100"
      height="100"
    />
    <h2 className="shoe-name">{props.product.product_name}</h2>
    <div className="shoe-info">
      Category:
      <br />
      <strong>{props.product.category}</strong>
    </div>
    <div className="shoe-info">
      Size:
      <br />
      <strong>{props.product.size}</strong>
    </div>
    <div className="shoe-info">
      Colour:
      <br />
      <strong>{props.product.colour}</strong>
    </div>
    <div className="initials">
      <strong>{props.product.customers_initials}</strong>
    </div>
    <button
      className="btn btn-remove"
      onClick={() =>
        props.handleProductRemove(props.product.id, props.product.product_name)
      }
    >
      <FontAwesomeIcon icon={faTimesCircle} />
    </button>
  </StyledProduct>
);

export default Product;
