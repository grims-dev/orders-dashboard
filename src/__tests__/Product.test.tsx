import { mount } from "enzyme";
import * as React from "react";
import toJSON from "enzyme-to-json";
import Product, { ProductUI } from "../components/Product";

function mountProduct(props: Partial<ProductUI> = {}) {
  const defaultProps: ProductUI = {
    position: 1,
    product: {
      id: 1,
      product_name: "Nike Air Max 95",
      category: "Category",
      size: "10",
      colour: "blue",
      status: "On the way",
      customers_initials: "CG",
    },
    handleProductRemove: jest.fn(),
  };

  return mount(<Product {...defaultProps} />);
}

describe("Product renders successfully", () => {
  it("renders without crashing", () => {
    const wrapper = mountProduct();
    const component = wrapper.find('[data-test="product"]');
    expect(toJSON(component)).toMatchSnapshot();
  });

  it("renders snapshot of product app", () => {
    const wrapper = mountProduct();
    expect(wrapper).toMatchSnapshot();
  });
});
