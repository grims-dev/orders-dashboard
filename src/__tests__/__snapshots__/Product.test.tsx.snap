// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product renders successfully renders snapshot of product app 1`] = `
<Product
  handleProductRemove={[MockFunction]}
  position={1}
  product={
    Object {
      "category": "Category",
      "colour": "blue",
      "customers_initials": "CG",
      "id": 1,
      "product_name": "Nike Air Max 95",
      "size": "10",
      "status": "On the way",
    }
  }
>
  <styled.div
    data-test="product"
  >
    <div
      className="sc-AxjAm jMVGkZ"
      data-test="product"
    >
      <div
        aria-hidden="true"
        className="status-border status-On the way"
      />
      <img
        alt="Nike Air Max 95"
        className="shoe-thumbnail"
        height="100"
        src=""
        width="100"
      />
      <h2
        className="shoe-name"
      >
        Nike Air Max 95
      </h2>
      <div
        className="shoe-info"
      >
        Category:
        <br />
        <strong>
          Category
        </strong>
      </div>
      <div
        className="shoe-info"
      >
        Size:
        <br />
        <strong>
          10
        </strong>
      </div>
      <div
        className="shoe-info"
      >
        Colour:
        <br />
        <strong>
          blue
        </strong>
      </div>
      <div
        className="initials"
      >
        <strong>
          CG
        </strong>
      </div>
      <button
        className="btn btn-remove"
        onClick={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f057",
                "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z",
              ],
              "iconName": "times-circle",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-times-circle fa-w-16 "
            data-icon="times-circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </FontAwesomeIcon>
      </button>
    </div>
  </styled.div>
</Product>
`;

exports[`Product renders successfully renders without crashing 1`] = `
Array [
  <styled.div
    data-test="product"
  >
    <div
      className="sc-AxjAm jMVGkZ"
      data-test="product"
    >
      <div
        aria-hidden="true"
        className="status-border status-On the way"
      />
      <img
        alt="Nike Air Max 95"
        className="shoe-thumbnail"
        height="100"
        src=""
        width="100"
      />
      <h2
        className="shoe-name"
      >
        Nike Air Max 95
      </h2>
      <div
        className="shoe-info"
      >
        Category:
        <br />
        <strong>
          Category
        </strong>
      </div>
      <div
        className="shoe-info"
      >
        Size:
        <br />
        <strong>
          10
        </strong>
      </div>
      <div
        className="shoe-info"
      >
        Colour:
        <br />
        <strong>
          blue
        </strong>
      </div>
      <div
        className="initials"
      >
        <strong>
          CG
        </strong>
      </div>
      <button
        className="btn btn-remove"
        onClick={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f057",
                "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z",
              ],
              "iconName": "times-circle",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-times-circle fa-w-16 "
            data-icon="times-circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </FontAwesomeIcon>
      </button>
    </div>
  </styled.div>,
  <div
    className="sc-AxjAm jMVGkZ"
    data-test="product"
  >
    <div
      aria-hidden="true"
      className="status-border status-On the way"
    />
    <img
      alt="Nike Air Max 95"
      className="shoe-thumbnail"
      height="100"
      src=""
      width="100"
    />
    <h2
      className="shoe-name"
    >
      Nike Air Max 95
    </h2>
    <div
      className="shoe-info"
    >
      Category:
      <br />
      <strong>
        Category
      </strong>
    </div>
    <div
      className="shoe-info"
    >
      Size:
      <br />
      <strong>
        10
      </strong>
    </div>
    <div
      className="shoe-info"
    >
      Colour:
      <br />
      <strong>
        blue
      </strong>
    </div>
    <div
      className="initials"
    >
      <strong>
        CG
      </strong>
    </div>
    <button
      className="btn btn-remove"
      onClick={[Function]}
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f057",
              "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z",
            ],
            "iconName": "times-circle",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-times-circle fa-w-16 "
          data-icon="times-circle"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </FontAwesomeIcon>
    </button>
  </div>,
]
`;
